det3d/datasets/argoverse/argoverse_common.py

```diff
@@ -0,0 +1,330 @@
+        self.record = record
+        self.boxes = None
+        self.options = {"BACKGROUND": 0, "VEHICLE": 1}
+        self.label_class = label
+        self.label_num = self.options[self.label_class]
+
+
+def load_all_clouds(dataset: str, log_id: str, timestamp_list: List[int]) -> Dict:
+    """Loads all point clouds from the Argoverse Tracking set
+
+    Args:
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+        timestamp_list: (list[int]) List of timestamps to load point clouds from
+    Returns:
+        cloud_dict: (dict) A dictionary with the timestamp as key and point cloud as value
+    """
+    cloud_dict = {}
+    file_path = os.path.join(dataset, log_id, "lidar")
+    for timestamp in timestamp_list:
+        point_cloud = ply_loader.load_ply(
+            os.path.join(file_path, f"PC_{timestamp}.ply")
+        )
+        cloud_dict[timestamp] = point_cloud
+    return cloud_dict
+
+
+def perform_SE3(cloud_dict: Dict, dataset: str, log_id: str) -> Dict:
+    """Performs the SE3 transformation on all point clouds
+
+    Args:
+        cloud_dict: (dict) A dictionary with the timestamp as key and point cloud as value
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+    Returns:
+        cloud_dict: (dict) A dictionary with the timestamp as key and transformed point cloud as value
+    """
+    for k, v in cloud_dict.items():
+        SE3 = get_city_SE3_egovehicle_at_sensor_t(int(k), dataset, log_id)
+        cloud_dict[k] = SE3.transform_point_cloud(v)
+    return cloud_dict
+
+
+def grids_group_and_SE3(cloud_dict: Dict, dataset: str, log_id: str, n_t: int) -> Dict:
+    """Groups five point clouds and performs the necessary SE3 transformation
+
+    Args:
+        cloud_dict: (dict) A dictionary with the timestamp as key and transformed point cloud as value
+        dataset: (str) A string pointing to the path of the Argoverse dataset
+        log_id: (str) The ID of the data collected
+        n_t: (int) Number of timestamps per group
+    Returns:
+        cloud_dict: (dict) A dictionary with the timestamp as key and five consecutive point clouds as value
+    """
+    timestamps = []
+    for key in sorted(cloud_dict.keys()):
+        timestamps.append(key)
+        if len(timestamps) == n_t:
+            t0_to_map_SE3 = get_city_SE3_egovehicle_at_sensor_t(
+                int(timestamps[0]), dataset, log_id
+            )
+            map_to_t0_SE3 = t0_to_map_SE3.inverse()
+            rotation = np.eye(3)
+            translation = np.array([XSize / 2, YSize / 2, 0])
+            t0_to_occ_SE3 = se3.SE3(rotation, translation)
+            grids = []
+            for ts in timestamps:
+                pc = cloud_dict[ts]
+                pc = map_to_t0_SE3.transform_point_cloud(pc)
+                pc = t0_to_occ_SE3.transform_point_cloud(pc)
```

Comment:
i would probably just compose these transforms in advance, using `.right_multiply_SE3()`